engine/src/main/java/org/terasology/engine/subsystem/headless/renderer/HeadlessWorldRenderer.java

```diff
@@ -300,4 +295,15 @@ public int compare(RenderableChunk o1, RenderableChunk o2) {
         }
     }
 
+	@Override
+	public RenderingConfig getRenderingConfig() {
```

Comment:
Have you looked into obtaining RenderingConfig and WorldProvider via injection instead? Have a look at the RenderNodes listed in WorldRendererImpl.initGraph() to see how it's done. It involves taking advantage of the @In annotation.